pytket/pytket/backends/backend_exceptions.py

```diff
@@ -34,15 +33,13 @@ class CircuitNotRunError(Exception):
 
     def __init__(self, handle: ResultHandle):
         super().__init__(
-            "Circuit corresponding to {0!r} ".format(handle)
-            + "has not been run by this backend instance."
+            f"Circuit corresponding to {handle!r} "
```

Comment:
Without the `!r` it will call the `str()` method on `handle`; with it it will call the `repr()` method. I don't know if these are different but I assume the `!r` was put there for a reason.